targets/TARGET_Ambiq_Micro/TARGET_Apollo3/device/i2c_api.c

```diff
@@ -193,4 +234,246 @@ const PinMap *i2c_slave_scl_pinmap(void)
+
+    Transaction.ui8Priority     = 1;        // High priority for now.
+    Transaction.ui32InstrLen    = 1;
+    if(rx == NULL){
+        Transaction.eDirection      = AM_HAL_IOM_TX;
+        char * ptr = (char *)tx;
+        Transaction.ui32Instr       = *ptr;
+        Transaction.ui32NumBytes    = tx_length;
+        ptr++;
+        Transaction.pui32TxBuffer   = (const uint32_t *)ptr;
+    }
+    else if(tx == NULL){
+        Transaction.eDirection      = AM_HAL_IOM_RX;
+        Transaction.ui32Instr       = *(char *)rx;
+        Transaction.ui32NumBytes    = rx_length;
+        Transaction.pui32RxBuffer   = (uint32_t *)rx;
+     }
+    Transaction.bContinue       = false;
+    Transaction.ui8RepeatCount  = 0;
+    Transaction.ui32PauseCondition = 0;
+    Transaction.ui32StatusSetClr = 0;
+    Transaction.uPeerInfo.ui32I2CDevAddr = address;
+    
+    void *pCallbackCtxt; 
+    bool * g_IOMDMAComplete; 
+    
+    if(obj->i2c.iom_obj.iom.inst == 0) {
+    pCallbackCtxt = pfnIOM0_DMA_Callback;
+    g_IOM0DMAComplete = false ; 
+    g_IOMDMAComplete = &g_IOM0DMAComplete;
+    _dma0_isr_handler =  handler;
+    }
+    else if(obj->i2c.iom_obj.iom.inst == 1) 
+    {
+    pCallbackCtxt = pfnIOM1_DMA_Callback;
+    g_IOM1DMAComplete = false ; 
+     g_IOMDMAComplete = &g_IOM1DMAComplete;
+     _dma1_isr_handler =  handler;
+    }
+    else if(obj->i2c.iom_obj.iom.inst == 2) 
+    {
+    pCallbackCtxt = pfnIOM2_DMA_Callback;
+    g_IOM2DMAComplete = false ; 
+     g_IOMDMAComplete = &g_IOM2DMAComplete;
+     _dma2_isr_handler =  handler;
+    }
+    else if(obj->i2c.iom_obj.iom.inst == 3) 
+    {
+    pCallbackCtxt = pfnIOM3_DMA_Callback;
+    g_IOM3DMAComplete = false ; 
+     g_IOMDMAComplete = &g_IOM3DMAComplete;
+     _dma3_isr_handler =  handler;
+    }
+    else if(obj->i2c.iom_obj.iom.inst == 4) 
+    {
+    pCallbackCtxt = pfnIOM4_DMA_Callback;
+    g_IOM4DMAComplete = false ; 
+     g_IOMDMAComplete = &g_IOM4DMAComplete;
+     _dma4_isr_handler =  handler;
+    }
+    am_hal_iom_nonblocking_transfer(obj->i2c.iom_obj.iom.handle, &Transaction,pCallbackCtxt, NULL); 
+    
+    #if 1
+    while (!(*g_IOMDMAComplete)){
+        //am_util_stdio_printf("# trans: %d\n", Transaction.ui32NumBytes );
+        uint32_t ui32Status;
+        if (!am_hal_iom_interrupt_status_get(_dma_i2c_obj->i2c.iom_obj.iom.handle, true, &ui32Status)){
+        //am_util_stdio_printf("dma Int.: %x!\n",ui32Status);
+        am_hal_iom_status_t psStatus;
+        am_hal_iom_status_get(_dma_i2c_obj->i2c.iom_obj.iom.handle, &psStatus);
+        //am_util_stdio_printf("dma state: \n %x \n %x \n %x \n %x \n %x  \n %x \n",
+        //psStatus.bStatIdle,
+        //psStatus.bStatCmdAct,
+        //psStatus.bStatErr,
+        //psStatus.ui32DmaStat,
+        //psStatus.ui32MaxTransactions,
+        //psStatus.ui32NumPendTransactions);
+        }
+    }
+    
+    #endif
+}
+
+uint32_t i2c_irq_handler_asynch(i2c_t *obj)
+{
+MBED_ASSERT(obj);
```

Comment:
I noticed lot of incosistent code. Would you run astyle to align this file?

Reply:
I did an astyle format.